module-3/lab-intro-to-ml/your-code/main.ipynb

```diff
@@ -12,7 +12,7 @@
   },
```

Comment:
- In this one "Next, count the number of rows of austin and missing_values.", try the following and check the differences between len and size methods. 
```Python
print(austin.size)
print(missing_values.size)
```
```Python
#Try this one : 
missing_values['PrecipitationSumInches'].size / austin['PrecipitationSumInches'].size
#instead of yours 
austin['PrecipitationSumInches'].isna().sum()/len(austin['PrecipitationSumInches'])
```
- In this question "In the cell below, convert the Date column values from string to numeric values using toordinal()" you can do something as simple as the following to get the 
from datetime import datetime as dt
austin_fixed['Date'] = austin_fixed['Date'].apply(lambda date_col: dt.strptime(date_col, '%Y-%m-%d').toordinal())
```
- In this one "Your y should be a subset of austin_fixed containing one column TempAvgF." try the following instead of writing down every column label and then train the SVM in order to check if the kernel works properly.
```Python
response_col_names = list(austin_fixed.columns.values)
response_col_names.remove('TempAvgF')
X = austin_fixed[response_col_names]
y = austin_fixed['TempAvgF']
```
- If you want to play with synthetic data for classification problems, scikit-learn has a function to generate synthetic datasets. https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_classification.html.
It's very useful to play with different type of classifiers (two label)
```Python
from sklearn.datasets import make_classification
```
- If you have some free time train both kernels rbf and linear to check the differences in the performance metrics. It would be also interesting to explore the confusion matrix and the precision score as other performance metrics for classifiers. 
- Gamma and C parameters for SVM (rbf and poly, not linear). The default in SVC is rbf. Try to train both a linear a
https://www.youtube.com/watch?v=m2a2K4lprQw
https://www.youtube.com/watch?v=joTa_FeMZ2s